Sources/AST/Environment/Environment.swift

```diff
@@ -114,53 +114,97 @@ public struct Environment {
   }
 
   // MARK: - Compatibility
-  func areArgumentsCompatible(source: [FunctionArgument],
-                              target: EventInformation,
+  func areArgumentsCompatible(source: EventInformation,
```

Comment:
Why swap the arguments here as well as at the call site? Did `source` and `target` not have meaning originally?